src/transformers/__init__.py

```diff
@@ -6541,6 +6542,7 @@
             T5_PRETRAINED_MODEL_ARCHIVE_LIST,
             T5EncoderModel,
             T5ForConditionalGeneration,
+            T5ForEncoderBasedSequenceClassification,
```

Comment:
Thanks for the suggestion. 

The existing T5Encoder has no decoder layers at all, but the model we are adding has a single decoder layer. IMHO it will be confusing to call our model a variant of the T5Encoder since it still has a decoder layer.